Readme.md

```diff
@@ -2,64 +2,66 @@
-                name
-            }
+```json
+{
+    // Response to the "introspection query" (see below). This path is resolved relative to the config file location.
+    "schemaFilePath": "../some/schema-file.json",
+    // List of regexes
+    "excludes": ["\\bsome-thing", "_test.jsx?$"],
+    // Options for type generation (see below)
+    "options": {
+        "scalars": {
+            "JSONString": "string"
         }
     }
-`;
-
-console.log(documentToFlowTypes(MyQuery, schema))
-/*
-export type SomeQueryResponseType = {|
-    human: ?{|
-        id: string,
-        name: ?string,
-        homePlanet: ?string,
-        friends: ?$ReadOnlyArray<?{|
-            name: ?string
-        |}>,
-    |}
-|};
-*/
+}
 ```
 
-If you already have a setup whereby you collect all of your graphql literals, that may be all you need!
+Then run from the CLI, like so:
 
-Otherwise, we provide a way to hook into jest to automatically collect your queries and generate the types.
+```bash
+$ graphql-flow path/to/config.json
+```
 
-## Options for `documentToFlowTypes`
+Files will be discovoered relative to the current working directory.
 
-```js
-{
-    // Use nullable types where the graphql type is nullable. Included for legacy compatability,
-    // will probably remove once the mobile repo no longer needs it.
+To specify what file should be checked, pass them in as subsequent cli arguments.
+
+## Options (for the cli 'options' config item, or when running from jest):
+
+```ts
+type Options = {
+    // These are from the `documentToFlowTypes` options object above
     strictNullability: boolean = true,
-    // Output `$ReadOnlyArray<>` instead of `Array<>`, for stricter flow typing. On by default.
     readOnlyArray: boolean = true,
-    // A mapping of custom scalar names to the underlying json representation.
     scalars: {[key: string]: 'string' | 'boolean' | 'number'}
+
+    // Specify an opt-in pragma that must be present in a graphql string source
+    // in order for it to be picked up and processed
+    // e.g. set this to `"# @autogen\n"` to only generate types for queries that
+    // have the comment `# @autogen` in them.
+    pragma?: string,
+    // Specify a pragma that will turn off `strictNullability` for that
+    // source file. e.g. `"# @autogen-loose\n"`.
+    loosePragma?: string,
+    // If neither pragma nor loosePragma are specified, all graphql documents
+    // that contain a query or mutation will be processed.
+
+    // Any graphql operations containing ignorePragma will be skipped
+    ignorePragma?: string,
 }
 ```
 
-## Using jest to do the heavy lifting:
+## Using from jest
+
+You can also use jest to do the heavy lifting, running all of your code and collecting queries
+by mocking out the `graphql-tag` function itself. This requires that all graphql operations are
+defined at the top level (no queries defined in functions or components, for example), but can
+support more complicated fragment includes.
```

Comment:
What additional kinds of fragment includes does running from Jest enable?

Reply:
Yeah, as we discussed in our meeting, dynamically-generated fragments would not work with the way we safelist operations. For that reason, I'd just as soon remove the mention that Jest supports "more complicated fragment includes". Alternatively, we could document exactly what is possible with Jest and why we think it's not a good idea to make use of the extra capability.